### PR TITLE
METAL-855: baremetal: remove ironic-inspector

### PR DESCRIPTION
**Pull request by dtantsur**
The inspection functionality is now provided by ironic proper.

Also fixes a few minor quirks in the baremetal config, such as:
- Passing htpasswd variables to BMO (it cannot use them)
- Creating auth-config files (only required for cross-service calls).

### CODE REVIEW

**Comment by honza** [comment]
It looks like inspection timeouts.

**Comment by dtantsur** [comment]
@honza this stuff is red herring probably. I guess there is some other place where I need to tell it not to use inspector.

File: data/data/bootstrap/baremetal/files/etc/containers/systemd/image-customization.container [hunk 1]
```diff
@@ -19,7 +19,10 @@ Volume=/opt/openshift:/opt/openshift:z,ro
 Environment="DEPLOY_ISO=/shared/html/images/ironic-python-agent.iso"
 Environment="DEPLOY_INITRD=/shared/html/images/ironic-python-agent.initramfs"
 Environment="IRONIC_BASE_URL=${IRONIC_BASE_URL}"
-Environment="IRONIC_INSPECTOR_BASE_URL=${IRONIC_BASE_URL}"
+# FIXME(dtantsur): remove IRONIC_INSPECTOR_BASE_URL once
+# https://github.com/openshift/openstack-ironic-python-agent/commit/6cd36a750f27d65a3f24a47fff0ebde9e21b57f6
+# makes it into our images.
+Environment="IRONIC_INSPECTOR_BASE_URL=${IRONIC_BASE_URL}:6385/v1/continue_inspection"
```
Comment (dtantsur) [inline data/data/bootstrap/baremetal/files/etc/containers/systemd/image-customization.container:25]: This approach won't work, we actually need https://github.com/openshift/ironic-agent-image/pull/110 and remove the option.

**Pull request merged**